app/reducers/entityReducers/article.js

```diff
@@ -24,14 +24,14 @@ export default (state, action) => {
     }
     case UPDATE_ARTICLE_SUCCESS: {
       const article = action.payload.getIn(['entities', 'article']).first();
-      const articleUrl = state.getIn(['article']).findKey((el) => el.get('id') === action.meta.articleId);
 
-      return state.setIn(['article', articleUrl], article);
+      return state.update('article', (articles) =>
+                          articles.map((v) =>
+                                       (v.get('id') === action.meta.articleId ? article : v)));
```

Comment:
Did you think about adding to `meta` of action a current URL and then just updating the state with the article from the server using that URL?